pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -0,0 +1,283 @@
+				"machine": fmt.Sprintf("%s/%s", namespace, "fakeMachine"),
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Node",
+		},
+		Status: corev1.NodeStatus{
+			Conditions: []corev1.NodeCondition{
+				{
+					Type:   corev1.NodeReady,
+					Status: nodeReadyStatus,
+				},
+			},
+		},
+	}
+}
+
+func machine(name string) *capiv1alpha1.Machine {
+	return &capiv1alpha1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"foo": "a",
+				"bar": "b",
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Machine",
+		},
+		Spec: capiv1alpha1.MachineSpec{},
+	}
+}
+
+func machineHealthCheck(name string) *healthcheckingv1alpha1.MachineHealthCheck {
+	return &healthcheckingv1alpha1.MachineHealthCheck{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "MachineHealthCheck",
+		},
+		Spec: healthcheckingv1alpha1.MachineHealthCheckSpec{
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"foo": "a",
+					"bar": "b",
+				},
+			},
+		},
+		Status: healthcheckingv1alpha1.MachineHealthCheckStatus{},
+	}
+}
+
+func TestHasMatchingLabels(t *testing.T) {
+	machine := machine("machine")
+	testsCases := []struct {
+		machine            *capiv1alpha1.Machine
+		machineHealthCheck *healthcheckingv1alpha1.MachineHealthCheck
+		expected           bool
+	}{
+		{
+			machine: machine,
+			machineHealthCheck: &healthcheckingv1alpha1.MachineHealthCheck{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "MatchingLabels",
+					Namespace: namespace,
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind: "MachineHealthCheck",
+				},
+				Spec: healthcheckingv1alpha1.MachineHealthCheckSpec{
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"foo": "a",
+							"bar": "b",
+						},
+					},
+				},
+				Status: healthcheckingv1alpha1.MachineHealthCheckStatus{},
+			},
+			expected: true,
+		},
+		{
+			machine: machine,
+			machineHealthCheck: &healthcheckingv1alpha1.MachineHealthCheck{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "NoMatchingLabels",
+					Namespace: namespace,
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind: "MachineHealthCheck",
+				},
+				Spec: healthcheckingv1alpha1.MachineHealthCheckSpec{
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"no": "match",
+						},
+					},
+				},
+				Status: healthcheckingv1alpha1.MachineHealthCheckStatus{},
+			},
+			expected: false,
+		},
+	}
+
+	for _, tc := range testsCases {
+		if got := hasMatchingLabels(tc.machineHealthCheck, tc.machine); got != tc.expected {
+			t.Errorf("Expected %t, got %t", tc.expected, got)
+		}
+	}
+}
+
+func TestIsHealthy(t *testing.T) {
+	nodeHealthy := node("healthy", true)
+	nodeUnhealthy := node("unhealthy", false)
+
+	if got := isHealthy(nodeHealthy); got != true {
+		t.Errorf("Expected true, got %t", got)
```

Comment:
Is this meant to be `%T`, or `%v`?

Reply:
is %t for booleans right https://golang.org/pkg/fmt/?